src/http/headers.cr

```diff
@@ -83,24 +86,31 @@ struct HTTP::Headers
   def includes_word?(key, word)
     return false if word.empty?
 
-    word = word.downcase
-    # iterates over all header values avoiding the concatenation
-    get?(key).try &.each do |value|
-      value = value.downcase
-      offset = 0
-      while true
-        start = value.index(word, offset)
-        break unless start
-        offset = start + word.size
-
-        # check if the match is not surrounded by alphanumeric chars
-        next if start > 0 && value[start - 1].ascii_alphanumeric?
-        next if start + word.size < value.size && value[start + word.size].ascii_alphanumeric?
-        return true
+    values = @hash[wrap(key)]?
+    case values
+    when Nil
+      false
+    when String
+      values.compare(word, case_insensitive: true) == 0
+    else
+      word = word.downcase
+      values.each do |value|
+        value = value.downcase
+        offset = 0
+        while true
+          start = value.index(word, offset)
+          break unless start
+          offset = start + word.size
+
+          # check if the match is not surrounded by alphanumeric chars
+          next if start > 0 && value[start - 1].ascii_alphanumeric?
+          next if start + word.size < value.size && value[start + word.size].ascii_alphanumeric?
```

Comment:
Shouldn't the surroundings be checked in the single-value branch as well?

Reply:
I think you are right. I thought multiple values in http headers were separates by commas, but it can be a single value with commas in it. I'll fix it.

Reply:
Fixed, thanks! The specs were passing because they create headers with the special hash syntax, which uses `[]=`, and apparently that was always turning the value into an array... now that's no longer the case which would make working with `HTTP::Headers` consume less memory in general (parsing of `HTTP::Request` uses `HTTP::Headers#add?` which doesn't create an array if needed, so this doesn't affect the benchmarks I showed before).

Reply:
Yes. This method doesn't look for values but words.